snapcraft/internal/lxd.py

```diff
@@ -304,24 +304,97 @@ def _ensure_container(self):
             check_call([
                 'lxc', 'config', 'set', self._container_name,
                 'raw.idmap', 'both {} 0'.format(os.getuid())])
+            # Remove existing device (to ensure we update old containers)
+            devices = self._get_container_status()['devices']
+            if self._project_folder in devices:
+                check_call([
+                    'lxc', 'config', 'device', 'remove', self._container_name,
+                    self._project_folder])
             check_call([
                 'lxc', 'start', self._container_name])
 
     def _setup_project(self):
         self._ensure_mount(self._project_folder, self._source)
 
+    def _get_container_address(self):
+        network = self._get_container_status()['state']['network']['eth0']
+        for address in network['addresses']:
+            if address['family'] == 'inet':
+                return address['address']
+        raise RuntimeError('No IP found for {}'.format(self._container_name))
+
     def _ensure_mount(self, destination, source):
         logger.info('Mounting {} into container'.format(source))
+        remote, container_name = self._container_name.split(':')
+        if remote != 'local':
+            return self._remote_mount(destination, source)
+
         devices = self._get_container_status()['devices']
         if destination not in devices:
             check_call([
                 'lxc', 'config', 'device', 'add', self._container_name,
                 destination, 'disk', 'source={}'.format(source),
                 'path={}'.format(destination)])
 
+    def _remote_mount(self, destination, source):
+        # Generate an SSH key and add it to the container's known keys
+        keyfile = 'id_{}'.format(self._container_name)
+        keyfile_path = os.path.join(self._tmp, keyfile)
+        if not os.path.exists(keyfile_path):
+            # Swallow output and allow exceptions to save stdout
+            check_output(['ssh-keygen', '-o', '-N', '', '-f', keyfile_path])
+        # os.sep needs to be `/` and on Windows it will be set to `\`
```

Comment:
I don't understand how this comment applies here. You're not using `os.sep`, right?